lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
         try:
             try:
-                self.inst = ref(cb.im_self)
+                self.inst = ref(cb.im_self, self._destroy)
```

Comment:
If I understand you correctly, by ``object`` you refer to ``CallbackRegistry``.  Proxy holds no hardrefs back to ``CallbackRegistry``, you see in ``_BoundMethodProxy.add_callback()`` I used a ``_BoundMethodProxy``, aka weakref back to ``CallbackRegistry``, only level one recursion used here (as we **don't** add a callback to the callback's callback), so no worries.

In the case where we lose all hardrefs to the object held in the ``Proxy`` object, the 3 element-cycle will run, and callback to all the callbacks still there (I added the try except here to deal with that).

In the case where we loose all hardrefs to the proxy object itself, then nothing happens because nothing needs to happen, it has no hardrefs to it so it just gets GC'ed like any other object.